activesupport/lib/active_support/cache.rb

```diff
@@ -622,6 +631,8 @@ def normalize_key(key, options = nil)
         #   namespace_key 'foo', namespace: -> { 'cache' }
         #   # => 'cache:foo'
         def namespace_key(key, options = nil)
+          raise InvalidCacheKeyError.new(key) if key.nil? || key.empty?
+
```

Comment:
Should we introduce a `validate_key` that subclasses can reuse? They may override or not use this method, losing the validity check.